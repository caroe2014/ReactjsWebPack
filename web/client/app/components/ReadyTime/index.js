// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import { Box } from 'rebass';
import styled from 'styled-components';
import get from 'lodash.get';
import i18n from 'web/client/i18n';
import moment from 'moment-timezone';

const ReadyTimeContainer = styled(Box)`
    text-align: center;
    &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const TextContainer = styled(Box)`
  color: ${props => props.theme.colors.primaryTextColor};
`;

const GetReadyTimeContainer = styled(Box)`
  color: ${props => props.theme.colors.primaryTextColor};
  margin-top: 3px;
`;

class ReadyTime extends Component {

  getReadyTime () {
    const readyTime = get(this.props, 'readyTime.etf')
      ? get(this.props, 'readyTime.etf.minutes')
      : get(this.props, 'readyTime.minutes');
    const minTime = get(this.props, 'readyTime.minTime.minutes');
    const maxTime = get(this.props, 'readyTime.maxTime.minutes');
    const scheduledTime = get(this.props, 'scheduledTime');
    let readyText = '_ _ _';
    if (!scheduledTime) {
      if (readyTime >= 0) {
        if (readyTime === 0) {
          readyText = i18n.t('CART_LESS_MINUTE');
        } else if (readyTime === 1) {
          readyText = i18n.t('CART_IN_A_MINUTE');
        } else {
          readyText = readyTime > 60
            ? i18n.t('CART_IN_ABS_MINUTES', { readyTime: this.getAbsoluteReadyTime(readyTime) })
            : i18n.t('CART_IN_N_MINUTES', { readyTime: readyTime });
        }
      } else if (minTime >= 0 && maxTime >= 0) {
        if (minTime === 0 && maxTime === 0) {
          readyText = i18n.t('CART_LESS_MINUTE');
        } else if (minTime === maxTime && minTime !== 0) {
          readyText = minTime > 60
            ? i18n.t('CART_IN_ABS_MINUTES', { readyTime: this.getAbsoluteReadyTime(minTime) })
            : i18n.t(minTime === 1 ? 'CART_IN_A_MINUTE' : ('CART_IN_N_MINUTES', { readyTime: minTime }));
        } else if (minTime === 0 && maxTime === 1) {
          readyText = i18n.t('CART_IN_A_MINUTE');
        } else if (minTime === 0 && maxTime > 1) {
          readyText = maxTime > 60
            ? i18n.t('CART_IN_ABS_MINUTES', { readyTime: this.getAbsoluteReadyTime(maxTime) })
            : i18n.t('CART_LESS_THAN_N_MINUTES', { maxTime: maxTime });
        } else {
          readyText = maxTime > 60
            ? i18n.t('CART_IN_BETWEEN_MINUTES',
              { minTime: this.getAbsoluteReadyTime(minTime),
                maxTime: this.getAbsoluteReadyTime(maxTime)})
            : i18n.t('CART_M_TO_N_MINUTES', { minTime: minTime, maxTime: maxTime });
        }
      }
      return readyText;
    } else {
      return scheduledTime;
    }
  };

  getAbsoluteReadyTime (etf) {
    const {orderTime, timeZone, cart} = this.props;
    const timeWithEtf = cart ? moment() : moment(orderTime);
    timeWithEtf.add(etf, 'minutes').locale('en-US').tz(timeZone);
    timeWithEtf.set({
      minute: Math.ceil(timeWithEtf.get('minute') / 5) * 5
    });
    return moment(timeWithEtf).locale('en-US').format('LT');
  }

  getReadyText () {
    const { scheduledTime, scheduledDay } = this.props;
    return scheduledTime ? scheduledDay
      ? i18n.t('CART_SCHEDULE_TEXT_WITH_DAY',
        { scheduledDay: moment().add(scheduledDay, 'days').format('dddd, MMMM D') })
      : i18n.t('CART_SCHEDULE_TEXT') : i18n.t('CART_READY_TEXT');
  }

  render () {
    const readyTime = this.getReadyTime();
    const { scheduledTime, scheduledDay } = this.props;
    return (
      <ReadyTimeContainer
        className='ready-time-container'
        tabIndex={0}
        aria-label={i18n.t(`${scheduledTime ? 'CART_SCHEDULE_TEXT' : 'CART_READY_TEXT'}`) + readyTime} >
        <TextContainer>
          {scheduledTime ? scheduledDay
            ? i18n.t('CART_SCHEDULE_TEXT_WITH_DAY', { scheduledDay: moment().add(scheduledDay, 'days').format('dddd, MMMM D') }) // eslint-disable-line max-len
            : i18n.t('CART_SCHEDULE_TEXT') : i18n.t('CART_READY_TEXT')}
        </TextContainer>
        <GetReadyTimeContainer><strong className='ready-time'>{readyTime}</strong></GetReadyTimeContainer>
      </ReadyTimeContainer>
    );
  }
}

export default ReadyTime;
