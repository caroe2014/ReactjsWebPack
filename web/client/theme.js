// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
