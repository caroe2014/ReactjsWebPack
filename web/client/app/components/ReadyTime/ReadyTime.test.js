// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import {
  mount,
  shallow,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Adapter from 'enzyme-adapter-react-16';
import {
  ThemeProvider
} from 'styled-components';
import theme from 'web/client/theme';

import ReadyTime from '.';
configure({
  adapter: new Adapter()
});

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: (key) => key };
    return Component;
  },
  Trans: 'div'
}));

describe('ReadyTime', () => {

  let readyTime = {
    minutes: -1
  };

  it('does a snapshot test', () => {
    const wrapper = mount(<ThemeProvider theme={theme}><ReadyTime /></ThemeProvider>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(mount(<ThemeProvider theme={theme}><ReadyTime /></ThemeProvider>).exists()).toBe(true);
  });

  it('should render scheduledTime when provided in props', () => {
    const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime scheduledTime={'testTime'}/></ThemeProvider>);
    expect(wrapper.dive().instance().getReadyTime()).toBe('testTime');
  });

  it('should render readytime as empty when value is incorrect', () => {
    const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
    expect(wrapper.dive().instance().getReadyTime()).toBe('_ _ _');
  });

  it('should render readytime less than a minute if ready time is 0', () => {
    readyTime = {
      minutes: 0
    };
    const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
    expect(wrapper.dive().instance().getReadyTime()).toBe('CART_LESS_MINUTE');
  });

  it('should render readytime in a minute if ready time is 1', () => {
    readyTime = {
      minutes: 1
    };
    const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
    expect(wrapper.dive().instance().getReadyTime()).toBe('CART_IN_A_MINUTE');
  });

  it('should render readytime in n minutes if ready time is greater than 1', () => {
    readyTime = {
      minutes: 5
    };
    const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
    expect(wrapper.dive().instance().getReadyTime()).toBe('CART_IN_N_MINUTES');
  });

  describe('When readytime etf is served with variance', () => {
    it('should render readytime less than a minute if ready time is 0', () => {
      readyTime = {
        'etf': {
          'periodType':

        { 'name': 'Minutes' },
          'minutes': 0,
          'fieldType':

        { 'name': 'minutes' }
        }
      };
      const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
      expect(wrapper.dive().instance().getReadyTime()).toBe('CART_LESS_MINUTE');
    });

    it('should render readytime in a minute if ready time is 1', () => {
      readyTime = {
        'etf': {
          'periodType':

        { 'name': 'Minutes' },
          'minutes': 1,
          'fieldType':

        { 'name': 'minutes' }
        }
      };
      const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
      expect(wrapper.dive().instance().getReadyTime()).toBe('CART_IN_A_MINUTE');
    });

    it('should render readytime in n minutes if ready time is greater than 1', () => {
      readyTime = {
        'etf': {
          'periodType':

        { 'name': 'Minutes' },
          'minutes': 3,
          'fieldType':

        { 'name': 'minutes' }
        }
      };
      const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
      expect(wrapper.dive().instance().getReadyTime()).toBe('CART_IN_N_MINUTES');
    });
  });

  describe('When readytime min max is served with variance', () => {
    it('should render readytime in less than a  minute if min and max are 0', () => {
      readyTime = {
        'minTime': {
          'periodType':

        { 'name': 'Minutes' },
          'minutes': 0,
          'fieldType':

        { 'name': 'minutes' }
        },
        'maxTime': {
          'periodType':

        { 'name': 'Minutes' },
          'minutes': 0,
          'fieldType':

        { 'name': 'minutes' }
        }
      };
      const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
      expect(wrapper.dive().instance().getReadyTime()).toBe('CART_LESS_MINUTE');
    });

    it('should render readytime in n minutes if min and max are same but greater than 0', () => {
      readyTime = {
        'minTime': {
          'periodType':

          { 'name': 'Minutes' },
          'minutes': 5,
          'fieldType':

          { 'name': 'minutes' }
        },
        'maxTime': {
          'periodType':

          { 'name': 'Minutes' },
          'minutes': 5,
          'fieldType':

          { 'name': 'minutes' }
        }
      };
      const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
      expect(wrapper.dive().instance().getReadyTime()).toEqual({'readyTime': 5});
    });

    it('should render readytime in a  minute if min is 0 and  max is 1', () => {
      readyTime = {
        'minTime': {
          'periodType':

          { 'name': 'Minutes' },
          'minutes': 0,
          'fieldType':

          { 'name': 'minutes' }
        },
        'maxTime': {
          'periodType':

          { 'name': 'Minutes' },
          'minutes': 1,
          'fieldType':

          { 'name': 'minutes' }
        }
      };
      const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
      expect(wrapper.dive().instance().getReadyTime()).toBe('CART_IN_A_MINUTE');
    });

    it('should render readytime in less than n minutes if min is 0 and max is greater than 0', () => {
      readyTime = {
        'minTime': {
          'periodType':

            { 'name': 'Minutes' },
          'minutes': 0,
          'fieldType':

            { 'name': 'minutes' }
        },
        'maxTime': {
          'periodType':

            { 'name': 'Minutes' },
          'minutes': 5,
          'fieldType':

            { 'name': 'minutes' }
        }
      };
      const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
      expect(wrapper.dive().instance().getReadyTime()).toBe('CART_LESS_THAN_N_MINUTES');
    });

    it('should render readytime in m to n minutes if min and max are greater than 0', () => {
      readyTime = {
        'minTime': {
          'periodType':

              { 'name': 'Minutes' },
          'minutes': 3,
          'fieldType':

              { 'name': 'minutes' }
        },
        'maxTime': {
          'periodType':

              { 'name': 'Minutes' },
          'minutes': 5,
          'fieldType':

              { 'name': 'minutes' }
        }
      };
      const wrapper = shallow(<ThemeProvider theme={theme}><ReadyTime readyTime={readyTime}/></ThemeProvider>);
      expect(wrapper.dive().instance().getReadyTime()).toBe('CART_M_TO_N_MINUTES');
    });
  });
});
