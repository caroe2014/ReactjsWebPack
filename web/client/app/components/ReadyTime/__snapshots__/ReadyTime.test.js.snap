// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadyTime does a snapshot test 1`] = `
.c1 {
  box-sizing: border-box;
}

.c0 {
  text-align: center;
}

.c0:focus {
  outline: none;
  box-shadow: none;
}

.c2 {
  margin-top: 3px;
}

<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <ReadyTime>
    <Styled(Box)
      aria-label="CART_READY_TEXT_ _ _"
      className="ready-time-container"
      tabIndex={0}
    >
      <Box
        aria-label="CART_READY_TEXT_ _ _"
        className="ready-time-container c0"
        tabIndex={0}
      >
        <div
          aria-label="CART_READY_TEXT_ _ _"
          className="ready-time-container c0 c1"
          tabIndex={0}
        >
          <Styled(Box)>
            <Box
              className=""
            >
              <div
                className="c1"
              >
                CART_READY_TEXT
              </div>
            </Box>
          </Styled(Box)>
          <Styled(Box)>
            <Box
              className="c2"
            >
              <div
                className="c2 c1"
              >
                <strong
                  className="ready-time"
                >
                  _ _ _
                </strong>
              </div>
            </Box>
          </Styled(Box)>
        </div>
      </Box>
    </Styled(Box)>
  </ReadyTime>
</ThemeProvider>
`;
